py/picca/io.py

```diff
@@ -1069,19 +1076,39 @@ def read_from_minisv_desi(in_dir, catalog, pk1d=None):
 
     data = {}
     num_data = 0
-
-    files_in = glob.glob(os.path.join(in_dir, "**/coadd-*.fits"),
+    if usesinglenights:
+        files_in = glob.glob(os.path.join(in_dir, "**/coadd-*.fits"),
                          recursive=True)
-    petal_tile_night = [
-        f"{entry['PETAL_LOC']}-{entry['TILEID']}-{entry['NIGHT']}"
-        for entry in catalog
-    ]
-    petal_tile_night_unique = np.unique(petal_tile_night)
-    filenames = []
-    for f_in in files_in:
-        for ptn in petal_tile_night_unique:
+        petal_tile_night = [
+            f"{entry['PETAL_LOC']}-{entry['TILEID']}-{entry['NIGHT']}"
+            for entry in catalog
+        ]
+        petal_tile_night_unique = np.unique(petal_tile_night)
+    
+        filenames = []
+        for f_in in files_in:
+          for ptn in petal_tile_night_unique:
             if ptn in os.path.basename(f_in):
                 filenames.append(f_in)
+    else:
+        if useall:
+            files_in = glob.glob(os.path.join(in_dir, "**/all/**/coadd-*.fits"),
+                         recursive=True)
+        else:
+            files_in = glob.glob(os.path.join(in_dir, "**/deep/**/coadd-*.fits"),
+                         recursive=True)
+        petal_tile = [
+            f"{entry['PETAL_LOC']}-{entry['TILEID']}"
+            for entry in catalog
+        ]
+        petal_tile_unique = np.unique(petal_tile)
```

Comment:
Why do we need this "unique" call here? Do we have expect more than one file for this combination of petal and tile?

Reply:
yes, this goes over the objects in the cat not over the files, only files that have objects in the cat will then be read from disk at all.